src/util/fields.py

```diff
@@ -881,9 +881,16 @@ def _convert_field_to_company_dependent(
         where_condition = "{0} IS NOT NULL"
     else:
         where_condition = cr.mogrify("{0} != %s", [default_value]).decode()
-    where_condition += format_query(cr, " AND {} IS NOT NULL", company_field)
 
-    using = format_query(cr, "jsonb_build_object({}, {{0}})", company_field)
+    if company_field:
+        where_condition += format_query(cr, " AND {} IS NOT NULL", company_field)
+        using = format_query(cr, "jsonb_build_object({}, {{0}})", company_field)
+    else:
+        using = format_query(
+            cr,
+            "(SELECT jsonb_object_agg(c.id::text, {}.{{0}}) FROM res_company c)",
+            table,
+        )
```

Comment:
The company dependent field has a btree not null index, since it optimizes for spasing columns.
So if the original value is equal to the fallback, we shouldn't build the jsonb data.